### ISSUE TITLE
CF Template fails with "Custom Resource failed to stabilize in expected time"

### ISSUE THREAD

**Issue by sammyraul** [awslabs/video-on-demand-on-aws#35]
I used the build-s3-dist.sh and created my own bucket with "bucket-<region name" and with respect to S3 everything is correct as it is properly able to fetch the contents.
But my template always fails here.
`The following resource(s) failed to create: [S3Config, Uuid, MediaConvertEndPoint]. . Rollback requested by user.`

Not sure what exactly is the problem, if I directly launch the template from AWS VOD link then it works correctly, only when I use the uploaded code from my S3 bucket this problem comes.
Any help will be highly appreciated.
Thanks

**Comment by ffiore1** [comment]
@sammyraul  
I had a similar issue. 

In my case I was using the bucket name inconsistently.
in one of the 2 commands below I added a version "v" identitifier which I did not add to the 2nd command.

that caused the CFN template to fail at the custom resource creation

./build-s3-dist.sh bucket-name v1.0.0
aws s3 sync dist/  bucket-name/video-on-demand-on-aws/1.0.0/

after the fix (add or remove the v) consistently that worked. 

At least, that worked for me .. hope that helps

**Comment by sammyraul** [comment]
@ffiore1 
Thanks for your reply. And how did you name your bucket is it "bucket-name-<region name>"
or just "bucket-name"?
I tried with v1, it doesn't work. I will try with v1.0.0

**Comment by ffiore1** [comment]
in the **first** command I just specified bucket-name
./build-s3-dist.sh  **bucket-name** **v1.0.0**

in the second command I specified all of it, so
aws s3 sync dist/  **bucket-name-region**/video-on-demand-on-aws/v1.0.0/

the final result or s3 bucket name is **bucket-name-region**, with all the zip files in the /video-on-demand-on-aws/**v1.0.0**/ subfolder (I used the v thing, but that is a my choice )

**Comment by dscpinheiro** [comment]
Hi @sammyraul,

Would you be able to tell us which commands you used?
As @ffiore1 mentioned, the version must be the same in both.
In addition to that, the _build-s3-dist_ command expects the bucket name without the region suffix, but the _aws s3 sync_ should include it.

For example
```
cd deployment/
./build-s3-dist.sh bucket 1.01
aws s3 sync dist/ s3://bucket-us-east-1/video-on-demand-on-aws/1.01/
```

**Comment by sammyraul** [comment]
@ffiore1 @dscpinheiro 
Thanks for your reply.  Here are the steps
1) I created an S3 bucket with name "bucket-region"
2) ./build-s3-dist.sh bucket v1.0.0
3) aws s3 sync dist/ bucket-region/video-on-demand-on-aws/v1.0.0/
Actually I think the exact path for S3 is not a problem as all the other Lambda functions are properly fetched and deployed. For example 
`SnsNotification	CREATE_COMPLETE
StepFunctions	CREATE_COMPLETE
Mediainfo	CREATE_COMPLETE`
etc.
It is taking real long time and failing for 3 resources S3Config, Uuid & MediaConvertEndPoint:
`
S3Config	DELETE_FAILED	Custom Resource failed to stabilize in expected time
MediaConvertEndPoint	DELETE_FAILED	Custom Resource failed to stabilize in expected time
Uuid	DELETE_FAILED	Custom Resource failed to stabilize in expected time
ROLLBACK_IN_PROGRESS	The following resource(s) failed to create: [S3Config, Uuid, MediaConvertEndPoint]. . Rollback requested by user.
S3Config	CREATE_FAILED	Resource creation cancelled
Uuid	CREATE_FAILED	Custom Resource failed to stabilize in expected time
MediaConvertEndPoint	CREATE_FAILED	Custom Resource failed to stabilize in expected time`

Can it be because of permissions in S3. Not sure.
And if you see the CF template, in all these resources
`MediaConvertEndPoint:
    Type: "Custom::LoadLambda"
    Properties:
      ServiceToken: !GetAtt CustomResource.Arn
      Resource: "EndPoint"`
The serviceToken is CustomResource that means it is waiting on response from CustomResource
that means CustomResource is not getting fetched from S3 which itself is surprising as other Lambda code is fetched and deployed. Not sure what's going on

**Comment by dscpinheiro** [comment]
@sammyraul I'll try to reproduce it tomorrow, but in the meantime, could you also check the contents of the CloudWatch log group of the custom resource?
It should be named in the following format: /aws/lambda/${AWS::StackName}-custom-resource

We do log any errors (https://github.com/awslabs/video-on-demand-on-aws/blob/master/source/custom-resource/index.js#L93), so hopefully we should see something in those log groups.

**Comment by sammyraul** [comment]
@dscpinheiro 
Thank you so much for the CloudWatch Log group pointer.

I saw `Unable to import module 'index': Error` in the log group for custom resource.

Looked like some problem while packaging into zip files. My node and npm looked incompatible.
Didn't realize as it is packaging the zip files correctly.
I uninstalled node and install version8 compatible version and it worked perfectly.
It worked with node v8.16.1 and npm 5.5.1. Maybe for other version it may also work but for me
it worked with above version.

That may be the reason that some resources waiting for Lambda to be up and running.

Thanks
